odoo/tools/view_validation.py

```diff
@@ -2,16 +2,28 @@
 
 import collections
 import logging
+import os
+
+from lxml import etree
+from odoo import tools
 
 _logger = logging.getLogger(__name__)
 
 
 _validators = collections.defaultdict(list)
-def valid_view(arch):
+def valid_view(arch, view_type):
     for pred in _validators[arch.tag]:
         if not pred(arch):
             _logger.error("Invalid XML: %s", pred.__doc__)
             return False
+    if view_type in ['calendar', 'diagram', 'gantt', 'graph', 'pivot', 'search', 'tree']:
```

Comment:
Why are you not using a regular validation function registered normally e.g.
```python
@validate('calendar', 'diagram', 'gantt', 'graph', 'pivot', 'search', 'tree')
def schema_valid(arch):
    # get RNG validator and do validation here
```
?